### PR TITLE
Markdown lint rule tweaks

### PR DESCRIPTION
**Pull request by jeremywiebe**
Linked PRs: N/A

## Changes
- Moves linting files into `/docs` and use symlinks to place them where needed
- Update `bin/lint-md` so that it can find the `remarkrc` file relative to itself
- Remove a bunch of duplicated rules in `remarkrc`
- Switch to using `no-duplicate-headings-in-section` instead of `no-duplicate-headings`

## How To Test
- Switch to this branch's version of the mobify code style
    - Update `package.json` to have `"mobify-code-style":"mboify/mobify-code-style#markdown-lint-rule-tweaks"`
    - Run `npm i`
- Run `node_modules/bin/lint-md docs/public/latest/` (assuming Astro repo)
- Note that there are no messages about duplicate headings

## Applicable Research Resources
- [`no-duplicate-headings.js`](https://github.com/wooorm/remark-lint/blob/master/packages/remark-lint/lib/rules/no-duplicate-headings.js)
- [`no-duplicate-headings-in-section.js`](https://github.com/wooorm/remark-lint/blob/master/packages/remark-lint/lib/rules/no-duplicate-headings-in-section.js)

## TODOs:
- [ ] +1
- [x] Updated README
- [x] Updated CHANGELOG
- [ ] (Other applicable TODOs)

### CODE REVIEW

**Comment by johnboxall** [comment]
Hey @jeremywiebe, I just realized this repo isn't hooked up to CI yet – I've fixed that, but locally the tests are failing for me, which is makes sense given that we've updated the linting rules with this PR.

Could you please update the test fixtures with updated input/expected output?

https://github.com/mobify/mobify-code-style/tree/develop/docs/test

You can test the result by running `npm test`. Thanks!

File: docs/remarkrc [hunk 1]
```diff
@@ -16,21 +16,14 @@
         "maximum-heading-length": 60,
         "maximum-line-length": 80,
         "no-consecutive-blank-lines": true,
-        "no-duplicate-headings": true,
```
Comment (johnboxall) [inline docs/remarkrc]: @jeremywiebe could update the test fixtures to include an example that will fail / pass this rule?

**Comment by johnboxall** [comment]
Thanks for the updates @jeremywiebe. My feedback around adding an extra example of a failing filename referenced in `bad.md` is non-blocking if you would like to move ahead. 👍

File: docs/remarkrc (hunk 1 above)
Comment (jeremywiebe) [inline docs/remarkrc]: Added.

File: docs/remarkrc [hunk 2]
```diff
@@ -16,21 +16,14 @@
...
+        "no-duplicate-headings-in-section": true,
         "no-file-name-articles": true,
-        "no-file-name-articles": true,
```
Comment (jeremywiebe) [inline docs/remarkrc]: Added a filename test.

**Pull request merged**